mapclassify/value_by_alpha.py

```diff
@@ -0,0 +1,507 @@
+    alpha_bins : mapclassify instance
+        Object of classified values used for alpha.
+    cmap : matplotlib.colors.Colormap
+        Colormap for the VBA layer
+    ax : matplotlib Axes instance, optional
+        Axes in which to plot the figure in multiple Axes layout. Default is None
+    rgb_label : str, optional
+        Label for the y-axis; the rgb variable.
+    alpha_label : str, optional
+        Label for the x-axis; the alpha variable.
+    min_alpha : float = 0.2
+        Minimum alpha threshold to prevent fully transparent masking.
+
+    Returns
+    -------
+    fig : matplotlip Figure instance
+        Figure of Value by Alpha heatmap
+    ax : matplotlib Axes instance
+        Axes in which the figure is plotted
+
+    """
+    # VALUES
+    rgba, legend_cmap = _value_by_alpha_cmap(
+        rgb_bins.yb, alpha_bins.yb, cmap=cmap, min_alpha=min_alpha
+    )
+    # separate rgb and alpha values
+    alpha = rgba[:, 3]
+    # extract unique values for alpha and rgb
+    alpha_vals = np.unique(alpha)
+    rgb_vals = legend_cmap(
+        (rgb_bins.bins - rgb_bins.bins.min())
+        / (rgb_bins.bins.max() - rgb_bins.bins.min())
+    )[:, 0:3]
+
+    # PLOTTING
+    if ax is None:
+        fig = plt.figure()
+        ax = fig.add_subplot(111)
+    else:
+        fig = ax.get_figure()
+
+    for irow, alpha_val in enumerate(alpha_vals):
+        for icol, rgb_val in enumerate(rgb_vals):
+            rect = patches.Rectangle(
+                (irow, icol),
+                1,
+                1,
+                linewidth=3,
+                edgecolor="none",
+                facecolor=rgb_val,
+                alpha=alpha_val,
+            )
+            ax.add_patch(rect)
+
+    values_alpha, x_in_thousand = format_legend(alpha_bins.bins)
+    values_rgb, y_in_thousand = format_legend(rgb_bins.bins)
+    ax.plot([], [])
+    ax.set_xlim([0, irow + 1])
+    ax.set_ylim([0, icol + 1])
+    ax.set_xticks(np.arange(irow + 1) + 0.5)
+    ax.set_yticks(np.arange(icol + 1) + 0.5)
+    ax.set_xticklabels(
+        [f"< {val:1.1f}" for val in values_alpha],
+        rotation=30,
+        horizontalalignment="right",
+    )
+    ax.set_yticklabels([f"$<${val:1.1f}" for val in values_rgb])
+
+    alpha_label = alpha_label if alpha_label else "alpha variable"
+    rgb_label = rgb_label if rgb_label else "rgb variable"
+
+    if x_in_thousand:
+        ax.set_xlabel(f"{alpha_label} ($10^3$)")
+    if y_in_thousand:
+        ax.set_ylabel(f"{rgb_label} ($10^3$)")
+    else:
+        ax.set_xlabel(alpha_label)
+        ax.set_ylabel(rgb_label)
+
+    ax.spines["left"].set_visible(False)
+    ax.spines["right"].set_visible(False)
+    ax.spines["bottom"].set_visible(False)
+    ax.spines["top"].set_visible(False)
+    return fig, ax
+
+
+############################################################
+#
+# stuff from ``splot/_viz_utils.py
+#
+############################################################
+
+
+def format_legend(values):
```

Comment:
This could rely on format tooling in https://github.com/pysal/mapclassify/blob/cb25d35d1ccb1867249ab61fcf5180c93f613bcd/mapclassify/classifiers.py#L84 for consistency.

Reply:
good call on this